File: leetcode/partitionLabels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        
        li = [0 for i in range(26)]
        
        for i in range(len(s)): li[ord(s[i]) - ord('a')] = i
        
        ans = []
        l = -1
        r = 0
        
        for i in range(len(s)):
            r = r if r > li[ord(s[i]) - ord('a')] else li[ord(s[i]) - ord('a')]
            if r == i:
                ans.append(r - l)
                l = i
        
        return ans
            

File: leetcode/partitionLabels.js
/**
 * @param {string} s
 * @return {number[]}
 */
 var partitionLabels = function(s) {
  let last = Array(26)
  let s2 = s.split('');
  
  s2.forEach((el, i) => last[el.charCodeAt(0) - 'a'.charCodeAt(0)] = i)
  
  let l = -1
  let r = 0
  let ans = []
  
  s2.forEach((el, i) => {
      r = (r > last[el.charCodeAt(0) - 'a'.charCodeAt(0)]) ? r : last[el.charCodeAt(0) - 'a'.charCodeAt(0)]
      if(r == i) {
          ans.push(r - l)
          l = i
      }
  })
  
  return ans
};